viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -251,12 +260,24 @@ export class MeasurementTool extends Cognite3DViewerToolBase {
       throw new Error('Measurement mode is not active, call enterMeasurementMode()');
     }
     this.cancelActiveMeasurement();
-    this._viewer.off('click', this._handlePointerClick);
+
+    this.restoreViewerClickHandlers();
+
     this._viewer.off('beforeSceneRendered', this._handleClippingPlanes);
     this._events.measurementEnded.fire();
     this._measurementMode = false;
   }
 
+  private restoreViewerClickHandlers(): void {
+    const triggerView = this._viewer.getEventHandlerTrigger('click');
+
+    // Make sure to include listeners added after the override so that they persist
+    const potentiallyAddedListeners = triggerView.getListeners().splice(1);
+
+    triggerView.replaceListeners([...this._cachedClickHandlers, ...potentiallyAddedListeners]);
+    this._cachedClickHandlers.splice(0);
+  }
```

Comment:
Not sure about the API here, maybe we can make it a little easier? If we just want to temporary disable all listeners why we expose their replacement? Can't we just make a `disableCurrentlyAdded` method or something similar. Then the code here will be much simpler, without caring about added listeners and other stuff.